modules/sfp_dnsdb.py

```diff
@@ -0,0 +1,333 @@
+            f"https://api.dnsdb.info/dnsdb/v2/lookup/{endpoint}/{queryType}/{query}",
+            timeout=self.opts["_fetchtimeout"],
+            useragent="SpiderFoot",
+            headers=headers,
+        )
+
+        if res["code"] == "429":
+            self.sf.error("You are being rate-limited by DNSDB", False)
+            self.errorState = True
+            return None
+
+        if res["content"] is None:
+            self.sf.info(f"No DNSDB record found for {query}")
+            return None
+
+        splittedContent = res["content"].strip().split("\n")
+        if len(splittedContent) == 2:
+            self.sf.info(f"No DNSDB record found for {query}")
+            return None
+
+        try:
+            records = []
+            for content in splittedContent:
+                records.append(json.loads(content))
+        except Exception as e:
+            self.sf.error(f"Error processing JSON response from DNSDB: {e}", False)
+            return None
+
+        return records[1:-1]
+
+    def emit(self, etype, data, pevent, notify=True):
+        evt = SpiderFootEvent(etype, data, self.__name__, pevent)
+        if notify:
+            self.notifyListeners(evt)
+        return evt
+
+    def isTooOld(self, lastSeen):
+        ageLimitTs = int(time.time()) - (86400 * self.opts["age_limit_days"])
+        if self.opts["age_limit_days"] > 0 and lastSeen < ageLimitTs:
+            self.sf.debug("Record found but too old, skipping.")
+            return True
+        return False
+
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if self.errorState:
+            return None
+
+        self.sf.debug(f"Received event, {eventName}, from {srcModuleName}")
+
+        if self.opts["api_key"] == "":
+            self.sf.error("You enabled sfp_dnsdb but did not set an API key!", False)
+            self.errorState = True
+            return None
+
+        if eventData in self.results:
+            self.sf.debug(f"Skipping {eventData}, already checked.")
+            return None
+        self.results[eventData] = True
+
+        responseData = set()
+        coHosts = set()
+
+        if eventName == "DOMAIN_NAME":
+            rrsetRecords = self.query("rrset", "name", eventData)
+            if rrsetRecords is None:
+                return None
+
+            for record in rrsetRecords:
+                record = record.get("obj")
+                if self.checkForStop():
+                    return None
+
+                if self.isTooOld(record.get("time_last")):
+                    continue
+
+                if record.get("rrtype") not in (
+                    "A",
+                    "AAAA",
+                    "MX",
+                    "NS",
+                    "TXT",
+                    "CNAME",
+                ):
+                    continue
+
+                self.emit("RAW_RIR_DATA", str(record), event)
```

Comment:
`RAW_RIR_DATA` should only be sent once, representing the full contents of the API call, not for every record returned from a single API call.